SynthDet/Assets/Scripts/Runtime/SynthDetScenario.cs

```diff
@@ -19,142 +20,102 @@ namespace SynthDet.Scenarios
+        static ColorRgbParameter s_SemanticColorParameter = new ColorRgbParameter();
+        
+        AssetLoadingStatus m_LoadingStatus = AssetLoadingStatus.InProgress;
         List<string> m_LabelStringsForAutoLabelConfig = new List<string>();
 
         /// <inheritdoc/>
         protected override bool isScenarioReadyToStart
         {
             get
             {
-                switch (m_LoadingStatus)
-                {
-                    case AssetLoadingStatus.Complete:
-                        return true;
-                    case AssetLoadingStatus.LoadingCatalog:
-                    {
-                        foreach (var handle in m_CatalogHandles)
-                        {
-                            if (!handle.IsDone)
-                                return false;
-                        }
-                        
-                        foreach (var handle in m_CatalogHandles)
-                        {
-                            if (handle.Status != AsyncOperationStatus.Succeeded)
-                            {
-                                Debug.LogError("Catalog download failed");
-                                m_LoadingStatus = AssetLoadingStatus.Failed;
-                                return false;
-                            }
-                        }
-                        
-                        LoadAndLabelPrefabs();
-                        m_LoadingStatus = AssetLoadingStatus.LoadingPrefabs;
-                        return false;
-                    }
-                    case AssetLoadingStatus.LoadingPrefabs:
-                    {
-                        lock (m_Prefabs)
-                        {
-                            if (m_Prefabs.Count < m_NumPrefabsToLoad)
-                                return false;
-                            SetupLabelConfigs();
-                            var randomizer = GetRandomizer<ForegroundObjectPlacementRandomizer>();
-                            m_Prefabs.Sort((prefab1, prefab2) => prefab1.name.CompareTo(prefab2.name));
-                            randomizer.prefabs = m_Prefabs.ToArray();
-                            m_LoadingStatus = AssetLoadingStatus.Complete;
-                            return true;
-                        }
-                    }
-                    case AssetLoadingStatus.Failed:
-                    {
-                        Quit();
-                        return false;
-                    }
-                    default:
-                        return false;
-                }
+                if (m_LoadingStatus == AssetLoadingStatus.Complete)
+                    return true;
+                if (m_LoadingStatus == AssetLoadingStatus.Failed)
+                    Quit();
+                return false;
             }
         }
 
         protected override void OnAwake()
         {
             base.OnAwake();
 
-            Caching.ClearCache();
```

Comment:
Could we keep this line please? The cache has led to some nasty debugging issues where we thought we were doing things right but in fact were not.

Reply:
Sure thing